contrib/android/get_apk_versioncode.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+import importlib.util
+import os
+import sys
+
+ARCH_DICT = {
+    "x86_64": "4",
+    "arm64-v8a": "3",
+    "armeabi-v7a": "2",
+    "x86": "1",
+}
+
+
+def get_electrum_version() -> str:
+    project_root = os.path.abspath(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+    version_file_path = os.path.join(project_root, "electrum", "version.py")
+    # load version.py; needlessly complicated alternative to "imp.load_source":
+    version_spec = importlib.util.spec_from_file_location('version', version_file_path)
+    version_module = version = importlib.util.module_from_spec(version_spec)
+    version_spec.loader.exec_module(version_module)
+    return version.ELECTRUM_VERSION
+
+
+def get_android_versioncode(*, arch_name: str) -> int:
+    version_code = 0
+    # add ELECTRUM_VERSION
+    app_version = get_electrum_version()
+    app_version_components = app_version.split('.')
+    assert len(app_version_components) == 3, f"version str expected to have 3 components, but got {app_version!r}"
+    for i in app_version_components:
+        version_code *= 100
+        version_code += int(i)
+    # add arch
+    arch_code = ARCH_DICT[arch_name]
+    assert len(arch_code) == 1
+    version_code *= 10
+    version_code += int(arch_code)
+    # compensate for legacy scheme
+    # note: up until version 4.5.5, we used a different scheme for version_code.
+    #       4_______________4_05_05_00
+    #       ^ android arch, ^ app_version (4.5.5.0)
+    # This offset ensures that all new-scheme version codes are larger than the old-scheme version codes.
+    offset_due_to_legacy_scheme = 45_000_000
+    version_code += offset_due_to_legacy_scheme
```

Comment:
> I'm not sure if offset is needed, we simply can make a major release (I know there's no breaking change, but it would help android versioning to be much simpler to reason about, and easier to predict). then first digit will be `5`, which is higher than previously `x86_64`'s `4`

As long as the major version number is a single digit (which will be true for the foreseeable future), this 45 million offset is equivalent to a simple "45" string prefix. I don't think that's too complicated.

I don't think we want to do a major version number bump just for this.

Reply:
hmm, are you sure you don't want to put room for patch? if patch ever goes to two digits (ex, `4..5.10`), it will fail.
currently for `4.5.5` it yields `45405054`

Reply:
it's actually fine, I ran some more examples and all looked fine to me, even though I thought it would not be fine